src/mesido/electricity_physics_mixin.py

```diff
@@ -149,14 +174,22 @@ def path_variables(self):
         variables.extend(self.__asset_is_switched_on_var.values())
         variables.extend(self.__storage_charging_var.values())
         variables.extend(self.__set_point_var.values())
+        options = self.energy_system_options()
+        if options["electrolyzer_efficiency"] == ElectrolyzerOption.LINEARIZED_THREE_LINES_EQUALITY:
+            for line in self.__electrolyzer_is_active_linear_segment_var.keys():
+                variables.extend(self.__electrolyzer_is_active_linear_segment_var[line].values())
 
         return variables
 
     def variable_is_discrete(self, variable):
         """
         All variables that only can take integer values should be added to this function.
         """
-
+        options = self.energy_system_options()
+        if options["electrolyzer_efficiency"] == ElectrolyzerOption.LINEARIZED_THREE_LINES_EQUALITY:
+            for line in self.__electrolyzer_is_active_linear_segment_var.keys():
+                if variable in self.__electrolyzer_is_active_linear_segment_var[line]:
```

Comment:
In line with the above statement to switch the order of the nested dict, first check if variable in the map before looping over the lines, prevents an unnecessary loop if the asset is not an electrolyzer with the equality setting.

And even only having the map as a nested dict and __electrolyzer_is_active_linear_segment_var as 'normal' dict would prevent the need for a loop.

Reply:
Changed